### ISSUE TITLE
503 Service Unavailable - Exception 

### ISSUE THREAD

**Issue by coolhome** [viceroypenguin/Going.Plaid#56]
I discovered that if plaid returns a 503 status code which produced this nasty error below:

```cs
System.Text.Json.JsonException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. Path: $ | LineNumber: 0 | BytePositionInLine: 0.
 ---> System.Text.Json.JsonReaderException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. LineNumber: 0 | BytePositionInLine: 0.
   at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes)
   at System.Text.Json.Utf8JsonReader.Read()
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   --- End of inner exception stack trace ---
   at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, JsonReaderException ex)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadCore[TValue](JsonReaderState& readerState, Boolean isFinalBlock, ReadOnlySpan`1 buffer, JsonSerializerOptions options, ReadStack& state, JsonConverter converterBase)
   at System.Text.Json.JsonSerializer.ReadAsync[TValue](Stream utf8Json, Type returnType, JsonSerializerOptions options, CancellationToken cancellationToken)
   at System.Net.Http.Json.HttpContentJsonExtensions.ReadFromJsonAsyncCore[T](HttpContent content, Encoding sourceEncoding, JsonSerializerOptions options, CancellationToken cancellationToken)
   at Going.Plaid.PlaidClient.ResponseParser.BuildResponse[TResponse](HttpResponseMessage response)
   at Going.Plaid.PlaidClient.ResponseParser.ParseResponseAsync[TResponse]()
``` 

I imagine the response body is empty - https://github.com/viceroypenguin/Going.Plaid/blob/030a02dda11a61001cd925e14c15af197054472c/src/Plaid/PlaidClient.cs#L169

**Comment by viceroypenguin** [comment]
Thank you for reporting. I did not expect that. I'll take a look this weekend to clear it up. Are you coming across 503's from Plaid very often?

**Comment by coolhome** [comment]
@viceroypenguin I have only seen this error in a short duration of time. I have a PR where I craft a Response object but no error and only status.

**Comment by viceroypenguin** [comment]
Addressed by commits: 6b0ca927a30670ec63af4253d38a8dde2b86cda1, a97343feaafdfc4b9b3fe029cc9f9c27156c05a3

**Comment by coolhome** [comment]
@viceroypenguin thank you! This should work :)
